pubsub/tests/system.py

```diff
@@ -488,6 +488,38 @@ def test_streaming_pull_max_messages(
         finally:
             subscription_future.cancel()  # trigger clean shutdown
 
+    def test_streaming_pull_subscriber_permissions_sufficient(
+        self, publisher, topic_path, subscriber, subscription_path, cleanup
+    ):
+
+        # Make sure the topic and subscription get deleted.
+        cleanup.append((publisher.delete_topic, topic_path))
+        cleanup.append((subscriber.delete_subscription, subscription_path))
+
+        # create a topic and subscribe to it
+        publisher.create_topic(topic_path)
+        subscriber.create_subscription(subscription_path, topic_path)
+
+        # TODO: A service account granting only the pubsub.subscriber role must
+        # be used
+        filename = "/some/service/account/file.json.json"
```

Comment:
@busunkim96 Let's talk offline on how to inject such service account into Kokoro environment, and how to do the same for the convenience of developers (we cannot commit the actual service account `.json` key file, of course).

Reply:
There is now a service account with just Pub/Sub subscriber in `${KOKORO_GFILE_DIR}/pubsub-subscriber-service-account.json`

You can set an environment variable in https://github.com/googleapis/google-cloud-python/blob/master/.kokoro/build.sh

I _think_ there may actually be a way to use the existing service account to create a short-lived credential with only the Pub/Sub subscriber role. See https://cloud.google.com/iam/docs/creating-short-lived-service-account-credentials#sa-credentials-permissions and https://google-auth.readthedocs.io/en/latest/reference/google.auth.impersonated_credentials.html#module-google.auth.impersonated_credentials

This would also be nicer for developers trying to run the tests (they don't have to worry about having a separate service account with the correct role)